pydoit.md

```diff
@@ -0,0 +1,138 @@
+doit
+====
+
+[doit on github](https://github.com/pydoit/doit)
+
+~~~sh
+pip3 install doit
+~~~
+
+What is it?
+-----------
+
+`doit` is a python-based workflow tool on file-level that lets you specify *actions* that connect *dependencies* to *target*, e.g. 
+~~~py
+def task_paper():
+    return {"targets" : ["paper.pdf"],
+            "file_dep": ["paper.tex", "img.png"],
+            "actions" : ["pdflatex paper.tex"]}
+
+def task_plot():
+    return {"targets" : ["img.png"],
+            "file_dep": ["plot.py", "simulation.py"],
+            "actions" : ["python3 simulation.py --output plot_data.vtu",
+                         "python3 plot.py --input plot_data.vtu"]}
+~~~
+
+You can define your workflows as chains of independent or connected tasks and calling `doit` in the command line will automatically run them in the correct sequence. So a `doit paper` will first run the `task_plot` to create the `img.png` and call `pdflatex` afterwards. Also, It will only rerun *actions*, if their *dependencies* change. That means that running `doit paper` again, will image in no *action*, as everything is up-to-date.
```

Comment:
Also, **i**t

Reply:
The skip flag is a nice work around for huge simulations. It solves the problem of manually removing parts of the workflow. Another interesting question from my point of view remains that is the question why I should redo huge simulations that someone else has checked in (and all dependencies are still not changes, so theoretically not need to redo the computation). It might be interesting so see how other tools are handling this (e.g. in snakemake it seems that they can handle computations on different compute servers).

Reply:
From a practical point of view, I see the point in skipping simulations. But the reproducibility IMO suffers. I also expect that to be the reason, why a workflow/reproducibility tool like pydoit does not have such a feature. 

Snakemake (apparently based on timestamps to decide reruns) has an _ancient_ flag for the *dependencies*. With that flag, Snakemake will trigger the workflow only if the *target* does not exist. Subsequent changes to *dependencies* will not trigger actions. But this is also not what we want.